tests/interfaces/test_tensorflow.py

```diff
@@ -575,13 +575,8 @@ def expand(self):
                 U3(p[0], p[1], p[2], wires=0)
                 qml.expval(qml.PauliX(0))
 
-            qtape = qtape.expand()

```

Comment:
Since expansion happens inside `execute`, we don't need to expand here.

Reply:
Wait, which change is this?

Reply:
are there implications to this?

Reply:
`trainable_params` are not longer set on the tape provided to the `execute` function, since they may only be set on an expanded version of the tape.

Adjoint differentiation will now be supported on more general tapes.  Currently, the `QubitDevice.adjoint_jacobian` substitutes `Rot` gates with their expansion, and raises errors for any other trainable operation without an operation derivative.

Since we expand to supported tapes in `qml.execute`, we can expand unsupported gates multiple times till we get something supported.  We don't need to pre-expand `tape` to have it supported by `adjoint_jacobian`.

We now can take the `adjoint_jacobian` of tapes with `U` gates.

Reply:
Nice! I think all of this should be added to the changelog